root/etc/docker/modify_daemon.json

```diff
@@ -3,21 +3,10 @@
 {{- $result := fromJson .chezmoi.stdin -}}
 
 {{- /* Delete features.buildkit as it is now the default */ -}}
-{{- /* https://github.com/twpayne/chezmoi/issues/2748 */ -}}
-{{- $parentKeyName := "features" -}}
-{{- $subKeyName := "buildkit" -}}
-{{- if hasKey $result $parentKeyName -}}
-{{-   $parentKey := get $result $parentKeyName -}}
-{{-   if hasKey $parentKey $subKeyName -}}
-{{-     $_ := unset $parentKey $subKeyName -}}
-{{-   end -}}
-{{-   $_ := set $result $parentKeyName $parentKey -}}
-{{-   if empty (get $result $parentKeyName) -}}
-{{-     $_ := unset $result $parentKeyName -}}
-{{-   end -}}
-{{- end -}}
+{{- $result := deleteValueAtPath "features.buildkit" $result -}}
 
 {{- /* If there is no output, chezmoi deletes the file */ -}}
+{{- $result := pruneEmptyDicts $result -}}
```

Comment:
Note that `pruneEmptyDicts` does not prune the top level dict, so if the top level dict is empty, you'll still get `{}` after calling `toPrettyJson`.

Reply:
Yeap, I figured. That makes sense. That's why I still check if the final result is empty, and call `toPrettyJson` only if it isn't.